API.md

```diff
@@ -0,0 +1,141 @@
+            "name": "containerjfr",
+            "namespace": "default",
+            "resourceVersion": "298899",
+            "uid": "f895e203-37e9-11ea-8866-52fdfc072182"
+        }
+    }
+}
+```
+
+## Creating a new Flight Recording
+
+To start a new recording, you will need to add a new recording request to the `FlightRecorder` in the `spec.recordingRequests` array. You can do this nicely on the command line with `oc edit`. The recording request must include the following:
+
+1. `name`: a string uniquely identifying the recording within that service
+2. `eventOptions`: an array of string options passed to Container JFR. The `"ALL"` special string can be used to enable all available events.
+3. `duration`: length of the requested recording as a [duration string](https://golang.org/pkg/time/#ParseDuration).
+
+```shell
+oc edit -o json flightrecorders/containerjfr
+```
+```json
+    "spec": {
+        "port": 9091,
+        "recordingRequests": [
+                {
+                        "name": "my-recording",
+                        "eventOptions": [ "jdk.SocketRead:enabled=true", "jdk.SocketWrite:enabled=true" ],
+                        "duration": "30s"
+                }
+        ]
+    },
+    "status": {
+        "recordings": [],
+```
+
+Once the operator has processed the request, it will be removed from the `spec.recordingRequests` array and a corresponding entry will appear in the `status.recordings` array. A recording is considered `active` if it is currently running.
+
+```shell
+$ oc get -o json flightrecorders/containerjfr
+```
+```json
+    "spec": {
+        "port": 9091,
+        "recordingRequests": []
+    },
+    "status": {
+        "recordings": [
+            {
+                "active": true,
+                "duration": "30s",
+                "name": "my-recording",
+                "startTime": "2020-01-23T15:55:04Z"
+            }
+        ],
+```
+
+## Downloading a Flight Recording
+
+Once the recording completes, the operator archives the recording and places a download link in the recording's status that you can then download with curl.
+
+```shell
+$ oc get -o json flightrecorders/containerjfr
+```
+```json
+    "spec": {
+        "port": 9091,
+        "recordingRequests": []
+    },
+    "status": {
+        "recordings": [
+            {
+                "active": false,
+                "downloadUrl": "https://containerjfr-default.apps-crc.testing:443/recordings/172-30-157-32_my-recording_20200123T155535Z.jfr",
+                "duration": "30s",
+                "name": "my-recording",
+                "startTime": "2020-01-23T15:55:04Z"
+            }
+        ],
+```
+
+You'll need to pass your bearer token with the curl request. (You may also need -k if your test cluster uses a self-signed certificate)
+```shell
+$ curl -k -H "Authorization: Bearer $(oc whoami -t)" https://containerjfr-default.apps-crc.testing:443/recordings/172-30-157-32_my-recording_20200123T155535Z.jfr my-recording.jfr
```

Comment:
`oc whoami -t | base64` I think

Reply:
Is Base64 encoding for the web server as well or just for commands over WebSockets? Using version 0.13.0, I only succeed without the encoding.

Reply:
Ah, you're right - whatever charset is used by some auth system's tokens is probably a permissible subset of what's allowed in HTTP headers, so I didn't add base64 encode/decode there. But now I'm questioning if that was a strong enough assumption. Anyway, your example reflects the actual current state of things, so it's good.